src/repository/VertexRepository.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2022 IROHA LAB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BaseEntityRepository } from '@irohalab/mira-shared/repository/BaseEntityRepository';
+import { VertexMap } from '../domains/VertexMap';
+import { Vertex } from '../entity/Vertex';
+
+export class VertexRepository extends BaseEntityRepository<Vertex> {
+    public async getVertexMap(jobId: string): Promise<VertexMap> {
+        const vertices = await this.find({ jobId });
+        if (vertices && vertices.length > 0) {
+            const verticesMap = {} as VertexMap;
+            vertices.forEach(v => {
+                verticesMap[v.id] = v;
+            });
+            return verticesMap;
+        } else {
+            return null;
```

Comment:
Is it safe to return null here?

Reply:
I think it should be OK since the empty object of Javascript is not evaluated as false. use null is easier for the caller to determine whether there is a vertex map returned.